sunpy/coordinates/wcs_utils.py

```diff
@@ -74,6 +142,19 @@ def solar_wcs_frame_mapping(wcs):
                              attrs[2] * u.m,
                              **kwargs)
 
+    # Read the observer out of obsgeo for ground based observers
+    if observer is None:
```

Comment:
Beyond the scope of this PR, but would be nice to put all the logic for getting an observer out of a WCS into its own function.